linkerd/app/inbound/src/http/tests.rs

```diff
@@ -207,3 +306,41 @@ fn hello_server(
         Ok(io::BoxedIo::new(client_io))
     }
 }
+
+#[tracing::instrument]
+fn connect_error() -> impl Fn(Remote<ServerAddr>) -> io::Result<io::BoxedIo> {
+    move |_| {
+        Err(io::Error::new(
+            io::ErrorKind::Other,
+            "server is not listening",
+        ))
+    }
+}
+
+#[tracing::instrument]
+fn timeout_server(
+    http: hyper::server::conn::Http,
+) -> Box<dyn FnMut(Remote<ServerAddr>) -> ConnectFuture + Send> {
+    Box::new(move |endpoint| {
+        let http = http.clone();
+        let span = tracing::info_span!("timeout_server", ?endpoint);
+        Box::pin(
+            async move {
+                tracing::info!("sleeping so that the proxy hits a dispatch timeout");
```

Comment:
i'm surprised this hits the dispatch timeout rather than the connect timeout, but i guess it does?

Reply:
I've addressed this by adding the `ConnectTimeout` and changing the test code to use `to_tcp_connect` which ensures the tower timeouts are wrapped with this new error.